app/models/campaign.rb

```diff
@@ -156,8 +156,9 @@ def valid_start_and_end_dates?
   end
 
   def set_slug
+    campaign_slug = slug.present? ? slug : title
     # Strip everything but unicode letters and digits, and convert spaces to underscores.
-    self.slug = title.downcase.gsub(/[^\p{L}0-9 ]/, '').tr(' ', '_') unless slug.present?
+    self.slug = campaign_slug.downcase.gsub(/[^\p{L}0-9 ]/, '').tr(' ', '_')
```

Comment:
Looks like this change is causing some test failures, probably in connection with the default campaign in the fixture.

Reply:
It is passing now. Should only normalize if it is not already valid.